tests/random/prng_test.go

```diff
@@ -59,116 +118,312 @@ func TestRead(t *testing.T) {
 			var seed [32]byte
 			copy(seed[:], crypto.Keccak256([]byte(tt.seedFrom)))
 
-			samples, state, err := prng.Sample(nil, seed, uint16(tt.bitsPerSample), uint16(tt.numSamples))
-			if err != nil {
-				t.Fatalf("Sample(%x, 8 bits, 1000 samples) error %v", seed, err)
-			}
+			for _, ii := range getImplementations(t) {
+				t.Run(ii.name, func(t *testing.T) {
 
-			t.Run("probabilistic tests", func(t *testing.T) {
-				var min, max, sum uint64
-				min = math.MaxUint64
-				for i, s := range samples {
-					if !s.IsUint64() {
-						t.Fatalf("sample[%d].IsUint64() got false; want true", i)
+					samples, err := ii.prng.Sample(nil, seed, uint16(tt.bitsPerSample), uint16(tt.numSamples))
+					if err != nil {
+						t.Fatalf("Sample(%x, 8 bits, 1000 samples) error %v", seed, err)
 					}
 
-					u := s.Uint64()
-					sum += u
-					if u < min {
-						min = u
-					}
-					if u > max {
-						max = u
+					var min, max, sum uint64
+					var onesCount int
+
+					numMax := (1 << tt.bitsPerSample) - 1
+					distMean := float64(numMax) / 2.
+					distVar := float64(numMax*numMax) / 12.
+
+					min = math.MaxUint64
+
+					var cusum, cusumMax int64
+
+					for i, s := range samples {
+						if !s.IsUint64() {
+							t.Fatalf("sample[%d].IsUint64() got false; want true", i)
+						}
+
+						u := s.Uint64()
+						sum += u
+						if u < min {
+							min = u
+						}
+						if u > max {
+							max = u
+						}
+
+						onesCount += bits.OnesCount64(u)
+
+						for j := uint64(0); j < tt.bitsPerSample; j++ {
+							z := 2*int64((u>>j)&1) - 1
+							cusum += z
+							abs := abs(cusum)
+							if abs > cusumMax {
```

Comment:
```Go
if a := abs(cuSum); a > cuSumMax {
  cuSumMax = a
}
```

This avoids shadowing `abs` and also keeps the scope as tight as possible. The short variable name is in keeping with the style guide in that it's proportional to the size of the scope across which it's used (2 lines).